src/log-pcap.c

```diff
@@ -1077,6 +1295,111 @@ static OutputInitResult PcapLogInitCtx(ConfNode *conf)
             }
             SCLogInfo("Using log dir %s", pl->dir);
         }
+
+        const char *compression_str = ConfNodeLookupChildValue(conf,
+                "compression");
+
+        PcapLogCompressionData *comp = &pl->compression;
+        if (compression_str == NULL || strcmp(compression_str, "none") == 0) {
+            comp->format = PCAP_LOG_COMPRESSION_FORMAT_NONE;
+            comp->buffer = NULL;
+            comp->buffer_size = 0;
+            comp->file = NULL;
+            comp->pcap_buf = NULL;
+            comp->pcap_buf_size = 0;
+            comp->pcap_buf_wrapper = NULL;
+        } else if (strcmp(compression_str, "lz4") == 0) {
+#ifdef HAVE_LIBLZ4
+            if (pl->mode == LOGMODE_SGUIL) {
+                SCLogError(SC_ERR_INVALID_YAML_CONF_ENTRY, "Compressed pcap "
+                        "logs are not possible in sguil mode");
+                exit(EXIT_FAILURE);
+            }
+            pl->compression.format = PCAP_LOG_COMPRESSION_FORMAT_LZ4;
+
+            /* Use SCFmemopen so we can make pcap_dump write to a buffer. */
+
+            comp->pcap_buf_size = sizeof(struct pcap_file_header) +
+                    sizeof(struct pcap_pkthdr) + PCAP_SNAPLEN;
+            comp->pcap_buf = SCMalloc(comp->pcap_buf_size);
+            if (comp->pcap_buf == NULL) {
+                SCLogError(SC_ERR_MEM_ALLOC, "SCMalloc failed: %s",
+                        strerror(errno));
+                exit(EXIT_FAILURE);
```

Comment:
We probably shouldn't exit here, but instead return an OutputInitResult of failure. And again, cleanup any allocations.

Reply:
I tried staying in line with the rest of the function, e.g. at line 1290,
```C
if (stat(pl->dir, &stat_buf) != 0) {
    SCLogError(SC_ERR_LOGDIR_CONFIG, "The sguil-base-dir directory \"%s\" " 
           "supplied doesn't exist. Shutting down the engine",
            pl->dir);
    exit(EXIT_FAILURE);
}
```
Or is there a special reason why it needs to exit in those cases? I don't mind making the change you requested, I just want to understand the code better.

Reply:
Sorry, I missed that. I do see 2 cases here. One is an error in configuration where we exit out, and can be fixed by the user. The other is a malloc error, which the user potentially can't fix, and perhaps we should fail more gracefully there? @victorjulien Any thoughts here on exiting?

Reply:
Generally we try to continue on malloc errors at runtime. At startup time they are allowed to be fatal.